src/frontend/tool_manager.rs

```diff
@@ -0,0 +1,61 @@
+use std::collections::HashMap;
+
+use super::ui::tool::{crop::Crop, Tool, ToolIdentifier};
+
+pub struct ToolManager {
+    tools: HashMap<ToolIdentifier, Box<dyn Tool>>,
+    active_tool: Option<ToolIdentifier>,
+}
+
+impl Default for ToolManager {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl ToolManager {
+    pub fn new() -> Self {
+        Self {
+            tools: Self::create_tools(),
+            active_tool: Some(ToolIdentifier::Crop),
+        }
+    }
+
+    fn create_tools() -> HashMap<ToolIdentifier, Box<dyn Tool>> {
+        let mut map: HashMap<ToolIdentifier, Box<dyn Tool>> = HashMap::new();
+
+        map.insert(ToolIdentifier::Crop, Crop::boxed());
+
+        map
+    }
```

Comment:
We can make use of the [`HashMap::from`](https://doc.rust-lang.org/std/collections/struct.HashMap.html#impl-From%3C%5B(K,+V);+N%5D%3E-for-HashMap%3CK,+V%3E) implementation:

```suggestion
    fn create_tools() -> HashMap<ToolIdentifier, Box<dyn Tool>> {
        HashMap::from([
            (ToolIdentifier::Crop, Crop::boxed()),
        ])
    }
```